src/public/modules/forms/admin/components/edit-logic.client.component.js

```diff
@@ -74,8 +75,9 @@ function editLogicComponentController($uibModal, FormFields) {
   }
 
   vm.deleteLogic = function (logicIndex) {
+    const logicIdToDelete = vm.myform.form_logics[logicIndex]._id
+    AdminFormService.deleteFormLogic(vm.myform._id, logicIdToDelete)
```

Comment:
Since this is a promise, should we use `$q.when` and only splice the logic on success?

Reply:
thanks, edited